### DISCUSSION TITLE
How to set up on windows

### DISCUSSION THREAD

**Discussion by TheReiko69** [MusicPlayerDaemon/MPD#1598] [Q&A]
Recently I wanted to start using MPD on windows because I thought it would be cool to do so. I have tried my best to look at the documentation of how you would go about setting up MPD to work on windows but I honestly cant wrap my head around it. The only thing I can find within the documentation itself is how to compile it for windows, not really much so of how to go about setting it up and using it. Which is what I would like to learn.

**Comment by MaxKellermann** [comment]
True, the manual is missing that part, and somebody[tm] should write something.
On Windows, you start MPD just like you'd do on Linux: start `mpd.exe`, optionally with the usual command-line arguments.

**Reply by TheReiko69** [reply]
Starting `mpd.exe` just throws me `exception: No configuration file found` unfortunately.

**Reply by MaxKellermann** [reply]
`mpd --help` shows a list of possible command-line options, e.g. there's the option `--no-config` which tells MPD to start with default configuration, without a configuration file.

**Reply by TheReiko69** [reply]
doing `mpd --no-config` ends up returning with `exception: config parameter 'log_file' not found`

**Reply by MaxKellermann** [reply]
Oh, that's a confusing error indeed. That happens only on Windows, because there's no default logging destination on Windows; but you can tell MPD to log to the console with `--stdout`, and that should fix this problem.
Please report a bug on this confusing behavior.

**Reply by TheReiko69** [reply]
adding `--stdout` to the command `mpd --no-config` gave me this output. still not working.

`decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
output: No 'audio_output' defined in config file
output: Successfully detected a jack audio device
exception: Failed to load libjack64.dll: The specified module could not be found.
: Unknown error`

I will make a bug report on this.

**Reply by rockymystic** [reply]
hopefully someone can detail out how to install on windows successfully, i tried couple of methods none of them worked

**Reply by PablexXXXX** [reply]
You can install lobjack64 from https://jackaudio.org/downloads/

**Comment by andzejsp** [comment]
has anyone figured this out?

**Comment by skidoo23** [comment]
KISS ;) Get precompiled mpd.exe and create mpd.conf - https://mpd.fandom.com/wiki/Windows_Compatibility#Native you need at least music_directory and audio_output.

**Comment by mcbp223** [comment]
I have a weird permission issue I think, when running mpd as a service it creates the log file but can't write in it, however when I run it from an administrative command prompt, it does output log data in the file. Any suggestion on folder permissions, or other location for the log file?

I'm testing on Windows 11, with precompiled x64 mpd.exe in `C:\Windows\` and mpd.conf in `C:\ProgramData\mpd\` with these entries:

```ini
music_directory		"C:\\Users\\pc\\Music"
playlist_directory	"C:\\ProgramData\\mpd\\playlists"
db_file			"C:\\ProgramData\\mpd\\database"
log_file		"C:\\Windows\\Logs\\mpd\\mpd.log"
state_file		"C:\\ProgramData\\mpd\\state"
sticker_file		"C:\\ProgramData\\mpd\\sticker.sql"

# An example of a winmm output (Windows multimedia API).
#
audio_output {
	type		"winmm"
	name		"My WinMM output"
##	device		"Digital Audio (S/PDIF) (High Definition Audio Device)" # optional
#		or
##	device		"0"		# optional
##	mixer_type	"hardware"	# optional
}

```

I gave full permissions to the `LOCAL SERVICE` account over the `C:\Windows\Logs\mpd\` folder.
The service was created in an administrative command prompt with
`sc create mpd binPath= "C:\Windows\mpd.exe" DisplayName= "Music Player Daemon"`
and it didn't need any wrapper on Windows 11, haven't tested it on other versions.

![mpd_reg](https://github.com/user-attachments/assets/8f41b23f-a737-431e-b96e-7445343f5e27)

**Reply by mateusfig** [reply]
I can't define the device to play. Nothing work. I tried list all devices with command below. I used the device's name and, afterward, DeviceID, but nothing work.

Get-CimInstance Win32_SoundDevice | Select-Object Name, DeviceID, Status

Anybody could help me? I'm trying use MPD on a customized Windows 2012 Server, only for audio playing, with minimal services running.

**Reply by mscrystalvoice** [reply]
install python and run attachment
[winMM_device.py](https://github.com/user-attachments/files/30009605/winMM_device.py)
[winMM_device_Output.py](https://github.com/user-attachments/files/30009607/winMM_device_Output.py)

PS/ the names are cropped correctly, for example "Hi-Fi Cable Input (VB-Audio Hi-"

> audio_output {
> type "winmm"
> name "WinMM output"
> #device "0"
> device "Динамики (Realtek High Definiti"
> }

**Reply by mateusfig** [reply]
Thank you in advance. I ditched the idea but I will try once again.

**Reply by wilsonletras** [reply]
@mateusfig I just followed this guide and works quite well here on two machines: https://github.com/zfphex/ncmpcpp-windows
